roseWong/assignments/mergeIntervals/lc56/lc56.py

```diff
@@ -0,0 +1,93 @@
+# Problem Statement #
+
+# Given a list of intervals, merge all the overlapping intervals to produce a list that has only mutually exclusive intervals.
+
+from __future__ import print_function
+
+
+class Interval:
+  def __init__(self, start, end):
+    self.start = start
+    self.end = end
+
+  def print_interval(self):
+    print("[" + str(self.start) + ", " + str(self.end) + "]", end='')
+
+
+def merge(intervals):
+  merged = []
+  if len(intervals) < 2:
+    return intervals
+# sort the intervals on the start time
+  intervals.sort(key=lambda x: x.start)
+  mergedIntervals = []
+  start = intervals[0].start
+  end = intervals[0].end
+  for i in range(1, len(intervals)):
+    interval = intervals[i]
+#     if interval.start <= end:  # overlapping intervals, adjust the 'end'
+#       end = max(interval.end, end)
+#     else:  # non-overlapping interval, add the previous internval and reset
+#       mergedIntervals.append(Interval(start, end))
+#       start = interval.start
+#       end = interval.end
+
+#   # add the last interval
+#   mergedIntervals.append(Interval(start, end))
+#   return mergedIntervals
+  return merged
```

Comment:
When I test this code, the intervals do not print. If you are struggling to create your own solution, try to reverse engineer the solution provided.